### ISSUE TITLE
Bug Report - Argument Error - sexp_processor - version 4.12.1

### ISSUE THREAD

**Issue by yra-wtag** [presidentbeef/brakeman#1362]
### Background

Brakeman version: 4.5.1
Rails version: 4.2.2
Ruby version: 2.3.8

### Issue
Hello, We found the following error in the latest version `4.12.1` of `sexp_processor`
We do not get the error in `4.12.0`
NB: This gem is a dependency of `pronto-brakeman` and `pronto-flay` where we are getting the errors. However we are almost certain that the issue is somehow related to this version. As changing the version number makes it work.

I think its some feature `sexp_processor` introduced which causes this.
```
/bundle/gems/sexp_processor-4.12.1/lib/sexp.rb:214:in `line': setting s(:call, s(:lvar, :abcd), :[], s(:lit, :sequence_number)).line nil (ArgumentError)
	from /bundle/gems/brakeman-4.5.1/lib/ruby_parser/bm_sexp.rb:43:in `deep_clone'
	from /bundle/gems/brakeman-4.5.1/lib/brakeman/processors/alias_processor.rb:170:in `process_call'
	from /bundle/gems/brakeman-4.5.1/lib/ruby_parser/bm_sexp_processor.rb:76:in `block in process'
	from /bundle/gems/brakeman-4.5.1/lib/ruby_parser/bm_sexp_processor.rb:113:in `in_context'
	from /bundle/gems/brakeman-4.5.1/lib/ruby_parser/bm_sexp_processor.rb:72:in `process'
	from /bundle/gems/brakeman-4.5.1/lib/brakeman/processors/alias_processor.rb:660:in `process_op_asgn1'
	from /bundle/gems/brakeman-4.5.1/lib/ruby_parser/bm_sexp_processor.rb:76:in `block in process'
	from /bundle/gems/brakeman-4.5.1/lib/ruby_parser/bm_sexp_processor.rb:113:in `in_context'
	from /bundle/gems/brakeman-4.5.1/lib/ruby_parser/bm_sexp_processor.rb:72:in `process'
	from /bundle/gems/brakeman-4.5.1/lib/brakeman/processors/lib/processor_helper.rb:13:in `block in process_all!'
	from (eval):3:in `map!'
	from (eval):3:in `map!'
	from /bundle/gems/brakeman-4.5.1/lib/brakeman/processors/lib/processor_helper.rb:11:in `process_all!'
	from /bundle/gems/brakeman-4.5.1/lib/brakeman/processors/alias_processor.rb:413:in `block in process_defn'
	from /bundle/gems/brakeman-4.5.1/lib/brakeman/processors/alias_processor.rb:423:in `block in meth_env'
	from /bundle/gems/sexp_processor-4.12.1/lib/sexp_processor.rb:452:in `scope'
	from /bundle/gems/brakeman-4.5.1/lib/brakeman/processors/alias_processor.rb:420:in `meth_env'
	from /bundle/gems/brakeman-4.5.1/lib/brakeman/processors/alias_processor.rb:412:in `process_defn'
	from /bundle/gems/brakeman-4.5.1/lib/ruby_parser/bm_sexp_processor.rb:76:in `block in process'
	from /bundle/gems/brakeman-4.5.1/lib/ruby_parser/bm_sexp_processor.rb:113:in `in_context'
	from /bundle/gems/brakeman-4.5.1/lib/ruby_parser/bm_sexp_processor.rb:72:in `process'
	from /bundle/gems/brakeman-4.5.1/lib/brakeman/processors/alias_processor.rb:50:in `process_safely'
	from /bundle/gems/brakeman-4.5.1/lib/brakeman/processors/library_processor.rb:37:in `process_defn'
	from /bundle/gems/brakeman-4.5.1/lib/ruby_parser/bm_sexp_processor.rb:76:in `block in process'
	from /bundle/gems/brakeman-4.5.1/lib/ruby_parser/bm_sexp_processor.rb:113:in `in_context'
	from /bundle/gems/brakeman-4.5.1/lib/ruby_parser/bm_sexp_processor.rb:72:in `process'
	from /bundle/gems/brakeman-4.5.1/lib/brakeman/processors/lib/processor_helper.rb:13:in `block in process_all!'
	from (eval):3:in `map!'
	from (eval):3:in `map!'
	from /bundle/gems/brakeman-4.5.1/lib/brakeman/processors/lib/processor_helper.rb:11:in `process_all!'
	from /bundle/gems/brakeman-4.5.1/lib/brakeman/processors/lib/module_helper.rb:54:in `handle_class'
	from /bundle/gems/brakeman-4.5.1/lib/brakeman/processors/library_processor.rb:25:in `process_class'
	from /bundle/gems/brakeman-4.5.1/lib/ruby_parser/bm_sexp_processor.rb:76:in `block in process'
	from /bundle/gems/brakeman-4.5.1/lib/ruby_parser/bm_sexp_processor.rb:113:in `in_context'
	from /bundle/gems/brakeman-4.5.1/lib/ruby_parser/bm_sexp_processor.rb:72:in `process'
	from /bundle/gems/brakeman-4.5.1/lib/brakeman/processors/library_processor.rb:21:in `process_library'
	from /bundle/gems/brakeman-4.5.1/lib/brakeman/processors/model_processor.rb:31:in `process_class'
	from /bundle/gems/brakeman-4.5.1/lib/ruby_parser/bm_sexp_processor.rb:76:in `block in process'
	from /bundle/gems/brakeman-4.5.1/lib/ruby_parser/bm_sexp_processor.rb:113:in `in_context'
	from /bundle/gems/brakeman-4.5.1/lib/ruby_parser/bm_sexp_processor.rb:72:in `process'
	from /bundle/gems/brakeman-4.5.1/lib/brakeman/processors/lib/processor_helper.rb:13:in `block in process_all!'
	from (eval):3:in `map!'
	from (eval):3:in `map!'
	from /bundle/gems/brakeman-4.5.1/lib/brakeman/processors/lib/processor_helper.rb:11:in `process_all!'
	from /bundle/gems/brakeman-4.5.1/lib/brakeman/processors/lib/module_helper.rb:54:in `handle_class'
	from /bundle/gems/brakeman-4.5.1/lib/brakeman/processors/model_processor.rb:35:in `process_class'
	from /bundle/gems/brakeman-4.5.1/lib/ruby_parser/bm_sexp_processor.rb:76:in `block in process'
	from /bundle/gems/brakeman-4.5.1/lib/ruby_parser/bm_sexp_processor.rb:113:in `in_context'
	from /bundle/gems/brakeman-4.5.1/lib/ruby_parser/bm_sexp_processor.rb:72:in `process'
	from /bundle/gems/brakeman-4.5.1/lib/brakeman/processors/lib/processor_helper.rb:13:in `block in process_all!'
	from (eval):3:in `map!'
	from (eval):3:in `map!'
	from /bundle/gems/brakeman-4.5.1/lib/brakeman/processors/lib/processor_helper.rb:11:in `process_all!'
	from /bundle/gems/brakeman-4.5.1/lib/brakeman/processors/lib/module_helper.rb:22:in `handle_module'
	from /bundle/gems/brakeman-4.5.1/lib/brakeman/processors/model_processor.rb:39:in `process_module'
	from /bundle/gems/brakeman-4.5.1/lib/ruby_parser/bm_sexp_processor.rb:76:in `block in process'
	from /bundle/gems/brakeman-4.5.1/lib/ruby_parser/bm_sexp_processor.rb:113:in `in_context'
	from /bundle/gems/brakeman-4.5.1/lib/ruby_parser/bm_sexp_processor.rb:72:in `process'
	from /bundle/gems/brakeman-4.5.1/lib/brakeman/processors/model_processor.rb:21:in `process_model'
	from /bundle/gems/brakeman-4.5.1/lib/brakeman/processor.rb:55:in `process_model'
	from /bundle/gems/brakeman-4.5.1/lib/brakeman/scanner.rb:302:in `process_model'
	from /bundle/gems/brakeman-4.5.1/lib/brakeman/scanner.rb:297:in `block in process_models'
	from /bundle/gems/brakeman-4.5.1/lib/brakeman/scanner.rb:311:in `block in track_progress'
	from /bundle/gems/brakeman-4.5.1/lib/brakeman/scanner.rb:308:in `each'
	from /bundle/gems/brakeman-4.5.1/lib/brakeman/scanner.rb:308:in `track_progress'
	from /bundle/gems/brakeman-4.5.1/lib/brakeman/scanner.rb:295:in `process_models'
	from /bundle/gems/brakeman-4.5.1/lib/brakeman/scanner.rb:57:in `process'
	from /bundle/gems/brakeman-4.5.1/lib/brakeman.rb:361:in `scan'
	from /bundle/gems/brakeman-4.5.1/lib/brakeman.rb:80:in `run'
	from /bundle/gems/pronto-brakeman-0.9.1/lib/pronto/brakeman.rb:13:in `run'
	from /bundle/gems/pronto-0.9.5/lib/pronto/runners.rb:20:in `block in run'
	from /bundle/gems/pronto-0.9.5/lib/pronto/runners.rb:13:in `each'
	from /bundle/gems/pronto-0.9.5/lib/pronto/runners.rb:13:in `run'
	from /bundle/gems/pronto-0.9.5/lib/pronto.rb:64:in `run'
	from /bundle/gems/pronto-0.9.5/lib/pronto/cli.rb:61:in `block in run'
	from /bundle/gems/pronto-0.9.5/lib/pronto/cli.rb:60:in `chdir'
	from /bundle/gems/pronto-0.9.5/lib/pronto/cli.rb:60:in `run'
	from /bundle/gems/thor-0.19.4/lib/thor/command.rb:27:in `run'
	from /bundle/gems/thor-0.19.4/lib/thor/invocation.rb:126:in `invoke_command'
	from /bundle/gems/thor-0.19.4/lib/thor.rb:369:in `dispatch'
	from /bundle/gems/thor-0.19.4/lib/thor/base.rb:444:in `start'
	from /bundle/gems/pronto-0.9.5/bin/pronto:6:in `<top (required)>'
	from /usr/local/bundle/bin/pronto:17:in `load'
	from /usr/local/bundle/bin/pronto:17:in `<main>'
```

**Comment by presidentbeef** [comment]
Hi @yra-wtag,

Yes, this was addressed in #1360.

You can avoid the issue by enforcing the older version of `sexp_processor` or you could try ensuring that Brakeman is loaded first, because Brakeman bundles its own dependencies.

**Comment by yra-wtag** [comment]
> You can avoid the issue by enforcing the older version of `sexp_processor` or you could try ensuring that Brakeman is loaded first, because Brakeman bundles its own dependencies.
@presidentbeef yup, we already did it. Thanks for the awesome work. Please let us know when you guys are releasing a new version which fixes this without any user interaction. So we can revert our changes and move to that version.

**Comment by presidentbeef** [comment]
This is fixed with the 4.6.0 release.
